src/js/components/SelectMultiple/SelectMultipleContainer.js

```diff
@@ -0,0 +1,614 @@
+        ) {
+          event.preventDefault(); // prevent submitting forms
+          selectOption(activeIndex)(event);
+        }
+      },
+      [
+        activeIndex,
+        selectOption,
+        options,
+        disabled,
+        disabledKey,
+        valueKey,
+        labelKey,
+      ],
+    );
+
+    const customSearchInput = theme.select.searchInput;
+    const SelectTextInput = customSearchInput || TextInput;
+    const selectOptionsStyle = theme.select.options
+      ? {
+          ...theme.select.options.box,
+          ...theme.select.options.container,
+        }
+      : {};
+
+    // handle when limit is reached
+    useEffect(() => {
+      if (value.length === limit) {
+        const newDisabled = [...disabledProp];
+        // disable everything that is not selected
+        const originallyDisabled = (index) => {
+          const option = allOptions[index];
+          let result;
+          if (disabledKey) {
+            result = applyKey(option, disabledKey);
+          } else if (Array.isArray(disabledProp)) {
+            if (typeof disabledProp[0] === 'number') {
+              result = disabledProp.indexOf(index) !== -1;
+            } else {
+              result =
+                disabledProp.indexOf(
+                  getOptionValue(index, options, valueKey || labelKey),
+                ) !== -1;
+            }
+          }
+          return result;
+        };
+        for (let i = 0; i < options.length; i += 1) {
+          if (!isSelected(i) && !originallyDisabled(i)) {
+            newDisabled.push(options[i]);
+          }
+        }
+        if (usingKeyboard)
+          setShowA11yLimit('Selected. Maximum selection limit reached.');
+        setDisabled(newDisabled);
+      } else {
+        if (usingKeyboard) setShowA11yLimit(undefined);
+        setDisabled(disabledProp);
+      }
+    }, [
+      isSelected,
+      value,
+      limit,
+      disabledProp,
+      allOptions,
+      disabledKey,
+      labelKey,
+      options,
+      usingKeyboard,
+      valueKey,
+    ]);
+
+    // reset showA11yLimit after announcement is read
+    useEffect(() => {
+      if (showA11yLimit !== undefined) {
+        setTimeout(() => {
+          setShowA11yLimit(undefined);
+        }, 2000);
+      }
+    }, [showA11yLimit]);
+
+    return (
+      <Keyboard
+        onEnter={onSelectOption}
+        onSpace={onSelectOption}
+        onUp={onPreviousOption}
+        onDown={onNextOption}
+        onKeyDown={onKeyDownOption}
+        onEsc={onClose}
+      >
+        <StyledContainer
+          ref={ref}
+          as={Box}
+          id={id ? `${id}__select-drop` : undefined}
+          dropHeight={dropHeight}
+          a11yTitle="Select dropdown"
+        >
+          {showSelectedInline ? (
+            <Box
+              direction="row"
+              justify="between"
+              flex={false}
+              pad={{ horizontal: 'small', top: 'xsmall' }}
+            >
+              <Box
```

Comment:
This whole return block feels a bit repetitive -- should this Box be built into SelectionSummary? Then depending on if `showSelectedInline` or not, the Box props could be altered. Seems like since SelectionSummary should present in a row, that the box defining the row should be part of it.

Reply:
```
let inlineContent = (
           <SelectionSummary
                  allOptions={allOptions}
                  clearRef={clearRef}
                  disabled={disabled}
                  disabledKey={disabledKey}
                  isSelected={isSelected}
                  labelKey={labelKey}
                  limit={limit}
                  onChange={onChange}
                  options={options}
                  search={search}
                  setActiveIndex={setActiveIndex}
                  value={value}
                  valueKey={valueKey}
                  boxProps={showSelectedInline 
                      ? { pad: { vertical: 'xsmall' },
                           direction: "row",
                           justify: "between",
                           gap: "small",
                           fill: "horizontal" }
                       : { .... }
                    }
             />
);

if (showSelectedInline) inlineContent = (
         <Box
              direction="row"
              justify="between"
              flex={false}
              pad={{ horizontal: 'small', top: 'xsmall' }}
            >
               {inlineContent}
               <Button
                icon={<FormUp />}
                onClick={onClose}
                a11yTitle="Close Select"
                plain
              />
         </Box>
);


... 

return ( ....
   <StyledContainer>
         {inlineContent}
         ....

Reply:
Good idea, I broke this out as suggested